### PR TITLE
Add dynamic and import note.

### PR DESCRIPTION
**Pull request by joiglifberg**
Add note about import required to be explicit and dynamic required to be top-level module.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 8.6s | 8.8s | ⚠️ +188ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.058 | 2.084 | ⚠️ +0.03 |
| / avg req/sec | 1214.9 | 1199.38 | ⚠️ -15.52 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.105 | 1.092 | -0.01 |
| /error-in-render avg req/sec | 2261.78 | 2288.52 | +26.74 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 10.3s | 10.3s | -5ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: 651b56764ea9493c58b784dc3fca9f9571645f4c

**Comment by balazsorban44** [comment]
A related discussion https://twitter.com/browniefed/status/1336906160780132358?s=19

File: docs/advanced-features/dynamic-import.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ export default Home
 
 `DynamicComponent` will be the default component returned by `../components/hello`. It works like a regular React Component, and you can pass props to it as you normally would.
 
+> **Note**: import() needs to be explicitly written without template strings. Furthermore the import() has to be inside the dynamic() call for us to be able to match webpack bundles / module ids to the specific dynamic() call. Also, you can't use dynamic() inside of rendering as it needs to be marked in the top level of the module.
```
Comment (timneutkens) [inline docs/advanced-features/dynamic-import.md:74]:
```suggestion
> **Note**: import() needs to be explicitly written without template strings. Furthermore the import() has to be inside the dynamic() call for Next.js to be able to match webpack bundles / module ids to the specific dynamic() call and preload them before rendering. You can't use dynamic() inside of React rendering as it needs to be marked in the top level of the module for preloading to work.
```

**Review by timneutkens — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 10.1s | 10.3s | ⚠️ +152ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.386 | 2.402 | ⚠️ +0.02 |
| / avg req/sec | 1047.74 | 1040.88 | ⚠️ -6.86 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.242 | 1.266 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2012.4 | 1974.86 | ⚠️ -37.54 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 12.1s | 11.8s | -303ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: f06b33457827cddc8a232bb885121bff205f5979

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 10.6s | 10.5s | -106ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.387 | 2.394 | ⚠️ +0.01 |
| / avg req/sec | 1047.32 | 1044.27 | ⚠️ -3.05 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.355 | 1.308 | -0.05 |
| /error-in-render avg req/sec | 1844.73 | 1911.66 | +66.93 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 12.5s | 12.5s | -40ms |
| nodeModulesSize | 82.4 MB | 82.4 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| 677f882d2ed8..0679.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-b992ce4..2479.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.1 kB | 59.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | joiglifberg/next.js patch-1 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: 94438cc071561e8926fc45ff96dbede31af6adc9

**Comment by timneutkens** [comment]
Thanks!

**Pull request merged**